### PR TITLE
Issue #3380714: Replace legacy group types with flexible groups in demo content

### PR DESCRIPTION
**Pull request by Kingdutch**
## Problem
As part of the group migration we're removing all the old group types. For local development and quick set-up we have a bunch of demo content, but this still uses the old group types. Ideally we really should be using our new flexible group types.

## Solution
Replace the old group types with the corresponding flexible group types in the demo content.

## Issue tracker
https://www.drupal.org/project/social/issues/3380714
Subtask https://getopensocial.atlassian.net/browse/PROD-27946

## Theme issue tracker
*[Required if applicable] Paste a link to the drupal.org theme issue queue item, either from [socialbase](https://www.drupal.org/project/socialbase) or [socialblue](https://www.drupal.org/project/socialblue). If any other issue trackers were used, include links to those too.*

## How to test
- [ ] Install Open Social with demo content enabled
- [ ] Ensure the demo content groups are properly created
- [ ] Ensure the groups can be edited and saved
- [ ] Ensure members and content can be added to the demo groups as expected.

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3691-htvxqclmlzogf8uux4phqrwnxhvvwd5p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/659eb3486b340e1f8836b5b5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3691-htvxqclmlzogf8uux4phqrwnxhvvwd5p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/659eb3486b340e1f8836b5b5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3691-htvxqclmlzogf8uux4phqrwnxhvvwd5p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/659eb3486b340e1f8836b5b5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3691-htvxqclmlzogf8uux4phqrwnxhvvwd5p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/659eb3486b340e1f8836b5b5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3691-htvxqclmlzogf8uux4phqrwnxhvvwd5p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/659eb3486b340e1f8836b5b5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3691-htvxqclmlzogf8uux4phqrwnxhvvwd5p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/659eb3486b340e1f8836b5b5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3691-htvxqclmlzogf8uux4phqrwnxhvvwd5p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/659eb3486b340e1f8836b5b5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3691-htvxqclmlzogf8uux4phqrwnxhvvwd5p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/659eb3486b340e1f8836b5b5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3691-htvxqclmlzogf8uux4phqrwnxhvvwd5p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/659eb3486b340e1f8836b5b5

**Review by nechai — APPROVED** [review]

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3691-htvxqclmlzogf8uux4phqrwnxhvvwd5p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/659eb3486b340e1f8836b5b5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3691-htvxqclmlzogf8uux4phqrwnxhvvwd5p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/659eb3486b340e1f8836b5b5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3691-htvxqclmlzogf8uux4phqrwnxhvvwd5p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/659eb3486b340e1f8836b5b5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3691-htvxqclmlzogf8uux4phqrwnxhvvwd5p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/659eb3486b340e1f8836b5b5

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3691-htvxqclmlzogf8uux4phqrwnxhvvwd5p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/659eb3486b340e1f8836b5b5

**Review by ribel — APPROVED** [review]
As we discussed all looks good here, so we can merge this PR to unblock other ones, related to the Group Migration.

File: modules/custom/social_demo/content/entity/group.yml [hunk 1]
```diff
@@ -7,7 +7,12 @@
   uid: 44fc3642-da2f-11e5-b5d2-0a1d41d68578
   created: -60 day|15:30
   label: Official Group Educational Assistance Association
-  type: open_group
+  type: flexible_group
+  field_flexible_group_visibility: community
```
Comment (ronaldtebrake) [inline modules/custom/social_demo/content/entity/group.yml:11]: Seems like the `field_flexible_group_visibility` isn't taken in to account during creation for any of the groups.
Screenshot 2024-01-25 at 08 38 57
Screenshot 2024-01-25 at 08 47 24
Screenshot 2024-01-25 at 08 47 46

**Pull request merged**